apis/python/src/tiledbsc/soma.py

```diff
@@ -174,3 +175,141 @@ def var_keys(self):
         An alias for `soma.var.ids()`.
         """
         return self.var.ids()
+
+    # ----------------------------------------------------------------
+    def dim_slice(self, slice_obs_ids, slice_var_ids) -> Dict:
+        """
+        Subselects the SOMA's obs, var, and X/data using the specified obs_ids and var_ids.
```

Comment:
This needs to use
https://anndata.readthedocs.io/en/stable/generated/anndata.AnnData.layers.html
in order to preserve _all_ layers of `X`, e.g. `data` and `rankit`

Reply:
done